osu.Game/Overlays/Dashboard/CurrentlyPlayingDisplay.cs

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Linq;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Screens;
+using osu.Game.Online.API;
+using osu.Game.Online.API.Requests;
+using osu.Game.Online.Spectator;
+using osu.Game.Screens.Play;
+using osu.Game.Users;
+using osuTK;
+
+namespace osu.Game.Overlays.Dashboard
+{
+    internal class CurrentlyPlayingDisplay : CompositeDrawable
+    {
+        private IBindableList<int> playingUsers;
+
+        private FillFlowContainer<UserPanel> userFlow;
+
+        [Resolved]
+        private SpectatorStreamingClient spectatorStreaming { get; set; }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            RelativeSizeAxes = Axes.X;
+            AutoSizeAxes = Axes.Y;
+
+            InternalChild = userFlow = new FillFlowContainer<UserPanel>
+            {
+                RelativeSizeAxes = Axes.X,
+                AutoSizeAxes = Axes.Y,
+                Spacing = new Vector2(10),
+            };
+        }
+
+        [Resolved]
+        private IAPIProvider api { get; set; }
+
+        // temporary, should be game-global but i don't want to add more manager classes for now.
+        private static readonly Dictionary<int, User> user_cache = new Dictionary<int, User>();
```

Comment:
I really don't like this. If you're going to hack it that's fine, but not as a `static`... Storing `User` objects to a static gives me all sorts of bad feelings.

Highly prefer either removing this and waiting for the API issue to be resolved or moving it into `OsuGame` temporarily.

Reply:
what's the reasoning, other than "bad feelings"? even at a game level it's still going to be roughly the same, just in a different place. just completely against static?

i'd like to do the same with case-when for instance, but kind of gave up on that in local usages.

Reply:
Bad feelings are the only reasons. What do you mean by "case-when"?